docs/reference/uds-core/release-overview.md

```diff
@@ -0,0 +1,114 @@
+
+## Release Process
+
+### Official Releases
+
+The release process is automated using [release-please](https://github.com/googleapis/release-please) and GitHub workflows:
+
+1. Changes are merged to the `main` branch through pull requests
+2. Release-please automatically determines the next version based on conventional commits
+3. A release PR is created with updated version numbers and CHANGELOG entries
+4. Once merged, a new version tag is created and the release workflow is triggered
+5. Release artifacts are built, tested, and published
+6. CVE scans are performed on the released artifacts
+7. Release notes are automatically generated from the CHANGELOG
+
+### Release Artifacts
+
+Each release includes:
+
+- Zarf packages for all UDS Core components
+- Pre-configured UDS bundles for demos and development (with [`uds-k3d`](https://github.com/defenseunicorns/uds-k3d))
+- Documentation updates as necessary
+
+## Testing Strategy
+
+UDS Core undergoes rigorous testing before each release.
+
+### Functionality Testing
+
+- **Operator Logic Tests**: Validate UDS Operator functionality and reconciliation logic
+- **UI Journey Tests**: End-to-end testing of user workflows across application interfaces
+- **API Validation Tests**: Verify API endpoints and responses function correctly
+- **Cross-Distribution Tests**: Validate compatibility with [supported Kubernetes distributions](/reference/uds-core/distribution-support)
+- **Upgrade Tests**: Verify smooth upgrades from previous versions
+
+### Security Testing
+
+- **Static Analysis**: CodeQL scans for code vulnerabilities
+- **Dependency Scanning**: Checks for vulnerabilities in dependencies
+- **Container Scanning**: Analyzes container images for security issues
+- **Policy Compliance**: Validates adherence to security policies
+
+### Production Validation
+
+Defense Unicorns maintains several environments running UDS Core that are updated shortly after each release, providing additional real-world validation.
+
+## Development Builds
+
+### Nightly Snapshots
+
+Automated builds from the latest `main` branch are created daily at 10:00 UTC:
+
+- Tagged as `snapshot-latest` on GitHub
+- Available as Zarf packages and UDS bundles in the [GitHub Packages repository](https://github.com/orgs/defenseunicorns/packages?tab=packages&q=uds%2Fsnapshots+repo%3Adefenseunicorns%2Fuds-core)
+- Each snapshot is tagged with a unique identifier combining date string + commit hash (for immutable, pinned references), while the most recent snapshots are also tagged with `latest` (a mutable tag that always points to the newest snapshot)
+- Useful for testing new features before official releases
+- **Not recommended for production use**
+
+### Feature Previews
+
+For significant new features or architectural changes, special snapshot builds may be created:
+
+- Available from feature branches or `main`
+- Provided for early feedback and validation
+
+## Upgrade Recommendations
+
+When upgrading UDS Core:
```

Comment:
We probably should mention following UDS Identity Config Upgrade Guide here: https://uds.defenseunicorns.com/reference/uds-core/idam/upgrading-versions/

Reply:
Great idea - added a link both in this release-overview and the existing upgrade doc.